drivers/vulkan/vulkan_context.cpp

```diff
@@ -203,29 +203,32 @@ VKAPI_ATTR VkBool32 VKAPI_CALL VulkanContext::_debug_report_callback(
 		case VK_DEBUG_REPORT_ERROR_BIT_EXT:
 			ERR_PRINT(debugMessage);
 			break;
+		default:
+			WARN_PRINT("UNKNOWN VALIDATION REPORT TYPE");
+			break;
 	}
 
 	return VK_FALSE;
 }
 
-VkBool32 VulkanContext::_check_layers(uint32_t check_count, const char *const *check_names, uint32_t layer_count, VkLayerProperties *layers) {
+VkBool32 VulkanContext::_check_layers(const uint32_t check_count, const char *const *check_names, const uint32_t layer_count, const VkLayerProperties *layers) const {
```

Comment:
Not sure if I see any purpose for adding `cont` to by-value `int` arguments, it's not doing anything, and it does not matter if they are.

Reply:
Yes, but we don't modify this parameter at this method, so it's logical to make constant.

Reply:
We don't use `const` for primitive type arguments either, so please remove it.

Reply:
> We don't use `const` for primitive type function arguments either, so please remove it. (We do use `const` for Object-derived function arguments when possible still.)

Alright, but why is this bad?

Reply:
It's fine to use this in your own projects, but we considered it redundant since we alreaedy have a `p_` prefix for function parameters.

By the way, it looks like the expected `p_` prefix for function parameters is missing here.

Reply:
> Roger that, but why is this bad?

It's not bad, but in this case it's mostly a style thing. `const` is important for pointers, references, and non-POD structs. But for primitive type arguments, the compiler is restricted by ABI and there is pretty much nothing to optimize. And it's not critical information for the developers, whether it's changed or not does not affect anything.